Fix document error. (#611) (#617)
crystal-lang/crystal-book: docs/syntax_and_semantics/literals/array.md

[1, "hello", 'x'] # => Array(Int32 | String | Char)
```

> An explicit type can be specified by immediately following the closing bracket with `of` and a type or union. This overwrites the inferred type and can be used for example to create an array that holds only some types initially but can accept other types later.

Reviewer: A union type is a type, so there's no need to mention this explicitly.

Reviewer: ```suggestion
An explicit type can be specified by immediately following the closing bracket with `of` and a type. This overwrites the inferred type and can be used for example to create an array that holds only some types initially but can accept other types later.
```

> An explicit type can be specified by immediately following the closing bracket with `of` and a type, each separated by ` | `. This overwrites the inferred type and can be used for example to create an array that holds only some types initially but can accept other types later.

Reviewer 2: ```suggestion
An explicit type can be specified by immediately following the closing bracket with `of` and a type or union. This overwrites the inferred type and can be used for example to create an array that holds only some types initially but can accept other types later.
```
